pkg/filters/builder/requestadaptor/requestadaptor.go

```diff
@@ -298,6 +319,28 @@ func (ra *RequestAdaptor) Handle(ctx *context.Context) string {
 	return ""
 }
 
+// adaptSpecWithTemplate this will override the one-by-one spec
+func (ra *RequestAdaptor) adaptSpecWithTemplate(tempSpec *Spec) {
+	if tempSpec.Method == "" {
+		tempSpec.Method = ra.spec.Method
+	}
+	if tempSpec.Header == nil {
+		tempSpec.Header = ra.spec.Header
+	}
+	if tempSpec.Body == "" {
+		tempSpec.Body = ra.spec.Body
+	}
+	if tempSpec.Host == "" {
+		tempSpec.Host = ra.spec.Host
+	}
+	if tempSpec.Decompress == "" {
+		tempSpec.Decompress = ra.spec.Decompress
+	}
+	if tempSpec.Compress == "" {
+		tempSpec.Compress = ra.spec.Compress
+	}
```

Comment:
`Compress` and `Decompress` should not be a part of the template.

Reply:
Maybe someone want to determine Compress or Decompress dynamically. And exposing it to the user is not bad, isn't it?

Reply:
No.
the template should only include fields of the request, `Compress`, `Decompress`, `sign` are configurations of the filter, they are different things.